openedx/core/djangoapps/programs/tasks.py

```diff
@@ -323,7 +325,9 @@ def _retry_with_custom_exception(username, reason, countdown):
     except Exception as exc:
         error_msg = f"Failed to determine program certificates to be awarded for user {student}: {exc}"
         LOGGER.exception(error_msg)
-        raise _retry_with_custom_exception(username=username, reason=error_msg, countdown=countdown) from exc
+        raise MaxRetriesExceededError(
+            f"Failed to award a program certificate to user {username}. Reason: {error_msg}"
```

Comment:
I know this isn't your sin, but since we're in here and changing things, can we update this message to possibly log the LMS User Id over username? It gets a lil bit of PII out of our logs. It looks like we've retrieved the User instance, so hopefully not too big of a deal? 🤞